### PR TITLE
added daa (diamond alignment) datatype

### PR DESCRIPTION
**Pull request by bernt-matthias**
Implemented DAA datatype for the new diamond wrapper (https://github.com/bgruening/galaxytools/pull/747). I adapted it from the DMND type and hope that I got the magic correct: 

Definition in the diamond sources seems to be here: https://github.com/bbuchfink/diamond/blob/984bc7ab07031491d95d256f35e620167ca53185/src/output/daa_file.h#L38

Careful check would be good since this is my first data type for Galaxy and I'm new to treating binary files.

### CODE REVIEW

File: lib/galaxy/datatypes/binary.py [hunk 1]
```diff
@@ -1756,6 +1756,37 @@ def sniff(self, filename):
...
+    >>> fname = get_test_fname( 'interval.interval' )
+    >>> DMND().sniff( fname )
+    False
+    """
+    file_ext = "dmnd"
+    edam_format = ""
+
+    def __init__(self, **kwd):
+        Binary.__init__(self, **kwd)
+
```
Comment (mvdbeek) [inline lib/galaxy/datatypes/binary.py]: The linter complains about the spaces on this line

File: lib/galaxy/datatypes/binary.py [hunk 2]
```diff
@@ -1756,6 +1756,37 @@ def sniff(self, filename):
...
+    def sniff(self, filename):
+        # The first 8 bytes of any daa file are 0x3c0e53476d3ee36b
+
+        try:
+            header = open(filename, 'rb').read(8)
+            if header == self._magic:
+                return True
+            return False
+        except Exception:
```
Comment (mvdbeek) [inline lib/galaxy/datatypes/binary.py]: I don't think this needs to be wrapped in try/except, the framework will treat an Exception as False anyway. It is probably better to use a `with` statement for  opening/closing the file though.

File: lib/galaxy/datatypes/binary.py (hunk 1 above)
Comment (bernt-matthias) [reply lib/galaxy/datatypes/binary.py]: okay

File: lib/galaxy/datatypes/binary.py (hunk 2 above)
Comment (bernt-matthias) [reply lib/galaxy/datatypes/binary.py]: OK. Also changed this in the DMND datatype from where I copy-pasted the code

**Comment by mvdbeek** [comment]
Thanks @bernt-matthias, I've opened a PR with some minor stuff [here](https://github.com/bernt-matthias/galaxy/pull/2), if you merge that I'll merge this!

**Review by mvdbeek — APPROVED** [review]

**Comment by bernt-matthias** [comment]
could you review the datatype and sniffer entries in the datatypes.conf? I'm not sure if this is correct. Also: is there a possibility to test https://github.com/bgruening/galaxytools/pull/747 against this branch? `planemo s --galaxy_source https://github.com/bernt-matthias/galaxy.git --galaxy_branch "topic/daa"` only gives me an empty webpage.

Sorry about the mess with the commits ... I still don't get why this is happening (usually when I rebase)

**Comment by bgruening** [comment]
@bernt-matthias can you rebase this against latest dev?

**Comment by bernt-matthias** [comment]
I did 

```
git checkout dev
git pull upstream dev
git rebase dev topic/daa
```

Maybe I should have done `git rebase upstream/dev topic/daa` .. but intuitively this should be the same.

**Comment by nsoranzo** [comment]
A simple `git pull --rebase upstream dev` should be enough.

**Comment by bernt-matthias** [comment]
OK. There were a few conflict which I could mostly skip. When trying to push I got

```
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

Maybe 1 day ... I will learn it :)

**Comment by nsoranzo** [comment]
@bernt-matthias That should be fine, if locally `git log -p` looks fine, you can just force push: `git push --force`

**Comment by bernt-matthias** [comment]
tata. cool. 

I would like to test this branch with the diamond update that I'm just developing https://github.com/bgruening/galaxytools/pull/747. Any comments on whats the best way to do this?

**Comment by nsoranzo** [comment]
`planemo test --galaxy_root /PATH/TO/YOUR/LOCAL/GALAXY/REPO /PATH/TO/GALAXYTOOLS/ tools/diamond/`

**Comment by mvdbeek** [comment]
@bernt-matthias this is ready, right ?

**Comment by bernt-matthias** [comment]
@nsoranzo thanks testing works. But I cant serve this way. In the browser I see that `static/scripts/libs/require.js` can not be loaded (indeed the file is absent in this branch)? 

I just want to make sure that daa upload and sniffing works. 

@mvdbeek  Apart from this I'm happy with the PR.

**Comment by dannon** [comment]
@bernt-matthias Not sure what might have happened to your local client files, but you should always be able to fix up the client by running a rebuild with `make client` or `make client-production`.

**Comment by bernt-matthias** [comment]
got it running. apparently planemo does not like a galaxy folder for which galaxy never was started. 

however, the datatype detection does not work, also for daa files dmnd is detected. I tried to add `log.debug` statements, but they are never printed on the galaxy log output (neither for DAA.sniff nor for DMND.sniff). 

just double checked that `read(8) == binascii.unhexlify("6be33e6d47530e3c")`

so, I don not know why it fails... any suggestion where I could dig?

**Comment by bernt-matthias** [comment]
just commited a fix for a typo in test the test specified in the DAA class. 

current code also contains the debug output for DAA and DMND snifferes which needs to be removed before merging

**Comment by bernt-matthias** [comment]
got it (another stupid typo). daa detection works now.

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Thanks all!

**Pull request merged**